lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -1196,15 +1196,28 @@
             "flags": "-mcpu=thunderx2t99"
           }
         ],
-        "clang": {
-          "versions": ":",
-          "flags": "-march=armv8-a -mcpu=generic"
-        }
+        "clang": [
+          {
+            "versions": ":6.9",
+            "family": "aarch64",
+            "flags": "-march={family} -mattr=+v8.1a,+fp-armv8,+neon,+crc,+crypto"
+          },
+          {
+            "versions": "7:8.9",
+            "family": "aarch64",
+            "flags": "-march={family} -mattr=+v8.1a,+fp-armv8,+aes,+neon,+sha2,+crc,+crypto"
+          },
+          {
+            "versions": "9:",
+            "family": "aarch64",
+            "flags": "-march={family} -mcpu=thunderx2t99"
+          }
+        ]
```

Comment:
Thanks @alalazo, but these options raise errors following:
```
[ogura@langtx202 bin]$ ./clang test.c -march=aarch64 -mattr=+v8.1a,+fp-armv8,+neon,+                                                           crc,+crypto
clang-8: error: unknown argument: '-mattr=+v8.1a,+fp-armv8,+neon,+crc,+crypto'
clang-8: error: the clang compiler does not support '-march=aarch64'
```
So, options for `llc` and `clang` seem different. I considered options for `clang` compiler commands. Currently I 
 will Check whether it works normally. 
```
diff --git a/lib/spack/llnl/util/cpu/microarchitectures.json b/lib/spack/llnl/util/cpu/microarchitectures.json
index 3b58899..e37327a 100644
--- a/lib/spack/llnl/util/cpu/microarchitectures.json
+++ b/lib/spack/llnl/util/cpu/microarchitectures.json
@@ -1196,10 +1196,16 @@
             "flags": "-mcpu=thunderx2t99"
           }
         ],
-        "clang": {
-          "versions": ":",
-          "flags": "-march=armv8-a -mcpu=generic"
-        }
+        "clang": [
+          {
+            "versions": "3.9:",
+            "flags": "-march=armv8.1-a+crc+crypto"
+          },
+          {
+            "versions": "5.0:",
+            "flags": "-mcpu=thunderx2t99"
+          }
+        ]
       }
     },
     "a64fx": {
@@ -1246,10 +1252,16 @@
             "flags": "-arch=armv8.2a+crc+aes+sh2+fp16+sve -msve-vector-bits=512"
           }
         ],
-        "clang": {
-          "versions": ":",
-          "flags": "-march=armv8-a -mcpu=generic"
-        }
+        "clang": [
+          {
+            "versions": "3.9:",
+            "flags": "-march=armv8.2-a+crc+crypto+fp16"
+          },
+          {
+            "versions": "5.0:",
+            "flags": "-march=armv8.2-a+crc+crypto+fp16+sve"
+          }
+        ]
       }
     },
     "arm": {
```
I am sorry if there is a lack of consideration here.

Reply:
Thanks @t-karatsu. You're completely correct. I think we need to revisit also the flags that are passed to `x86-64`. I tried to disassemble some vectorizable code and it seems `clang` does not give the same meaning to flags that are also used by its assembler. I'll try to see if I can find some reference for it and eventually post here what I'll find.  @tgamblin FYI

Reply:
For reference, when I researched this I couldn't find a good source of information. Even `clang` manuals suggest options that are either not recognized on command line or don't work as described. The best source of information I could find at the time was [this SO answer](https://stackoverflow.com/a/35061913/771663).

Reply:
@becker33 I think you might also be interested in the discussion above.

Reply:
Further update: I found [this discussion](https://groups.google.com/forum/#!topic/llvm-dev/b1ckC_hini4) on how clang opt and llc relate to each other (and goes on a 4 stage manual build to use them). There's also [this other question](https://groups.google.com/forum/#!topic/llvm-dev/0f6kPmTo6OM) that looks exactly what we need, but has no answer. 

I'll continue looking, in the meanwhile I posted a message to cfe users - let's see if something comes out from there.

Reply:
> 
> 
> For reference, when I researched this I couldn't find a good source of information.

How about clang preprocessor test code?
    https://github.com/llvm/llvm-project/blob/master/clang/test/Preprocessor/aarch64-target-features.c

Reply:
@tkameyama @t-karatsu I got access today to a `thunderx2` machine. Will try out the flags there. 

> How about clang preprocessor test code?

That should work thanks. The front-end driver seems consistent with GCC in terms of flags which is also what is briefly written [here](https://clang.llvm.org/docs/DriverInternals.html). Will update this PR asap and ping you when done.